src/components/Profile/StakingTokens.tsx

```diff
@@ -0,0 +1,229 @@
+      proposalId: stake.index === -1 ? null : stakes[stake.index].proposalId,
+      proposalName:
+        stake.index === -1 ? 'Others' : stakes[stake.index].proposalName,
+    },
+    percentage: stake.percentage,
+  }))
+}
+
+const getStakeItemsPerGarden = (
+  stakes: Array<TransformedStakeType> | null,
+  garden: string
+) => {
+  const items: Array<any> = []
+
+  stakes?.map((stake) => {
+    if (stake?.item.gardenId === garden) {
+      items.push({
+        item: stake?.item,
+        percentage: Math.round(stake?.percentage) ?? '0',
+      })
+    }
+  })
+
+  return items
+}
+
+type Garden = { address: string; name: string }
+
+const getGardenNameByGardens = (
+  gardens: Array<Garden>,
+  gardenAddress: string
+) => {
+  const garden = gardens.find(
+    (garden: Garden) => garden.address.toLowerCase() === gardenAddress
+  )
+
+  return garden?.name ?? ''
+}
+
+const getStakesByGarden = (
+  gardensMetadata: Array<Garden>,
+  stakes: Array<TransformedStakeType> | null
+): Array<{
+  garden: string
+  gardenName: string
+  items: Array<TransformedStakeType>
+}> => {
+  if (stakes === null) return []
+
+  const gardens: Array<any> = []
+  const stakesPerGarden: Array<any> = []
+
+  // get all unique gardens
+  stakes.map((stake: TransformedStakeType) => {
+    const gardenId = stake?.item?.gardenId
+    if (gardenId && !gardens.includes(gardenId)) {
+      gardens.push(gardenId)
+    }
+  })
+
+  gardens.map((garden) => {
+    stakesPerGarden.push({
+      garden,
+      gardenName: getGardenNameByGardens(gardensMetadata, garden),
+      items: getStakeItemsPerGarden(stakes, garden),
+    })
+  })
+
+  return stakesPerGarden
+}
+
+function StakingTokens({ myStakes }: StakingTokensProps) {
+  const { gardensMetadata } = useGardens()
+  const theme = useTheme()
+  const { below } = useViewport()
+  const compact = below('large')
+
+  const myActiveTokens = useMemo(() => {
+    if (!myStakes) {
+      return new BigNumber('0')
+    }
+
+    return myStakes.reduce((accumulator, stake) => {
+      return accumulator.plus(stake.amount)
+    }, new BigNumber('0'))
+  }, [myStakes])
+
+  const stakes = useMemo(() => {
+    if (!myStakes) {
+      return null
+    }
+    return displayedStakes(myStakes, myActiveTokens)
+  }, [myStakes, myActiveTokens])
+
+  if (myActiveTokens.eq(0)) {
+    return null
+  }
+
+  const colors = [theme.green, theme.red, theme.purple, theme.yellow]
+  const stakesByGarden = getStakesByGarden(gardensMetadata, stakes)
+
+  return (
+    <Box heading="Supported proposals" padding={3 * GU}>
+      <p>Active token distribution</p>
+      {stakesByGarden.map(({ garden, gardenName, items }) => (
+        <div
+          key={garden}
+          css={`
+            margin-top: 25px;
+          `}
+        >
+          <Distribution
+            colors={colors}
+            heading={
+              <p
+                css={`
+                  margin-bottom: 10px;
+                  font-size: 16px;
```

Comment:
What do you think of giving the Garden name a bit more style? For example:
```
<Tag mode="identifier">
	{gardenName ?? shortenAddress(garden)}
</Tag>

```

Reply:
I think we could aim for something diferente since if we have a lot of active stakes and a lot of gardens it would be a lot of scrolling to do. We could have a toggle or modal for this. But as I suggested this could be done in a next PR. For this moment I will add the Tag element ✌️

Reply:
WDYT?
<img width="282" alt="Captura de ecrã 2022-03-26, às 16 12 42" src="https://user-images.githubusercontent.com/13184907/160248068-f435842c-9f51-4e4a-8126-c2f1f9bc72d4.png">

Reply:
Looks great!